torch/autograd/__init__.py

```diff
@@ -265,7 +267,18 @@ def grad(
             cliffs. Please use ``torch._C._debug_only_display_vmap_fallback_warnings(True)``
             to show any performance warnings and file an issue on github if warnings exist
             for your use case. Defaults to ``False``.
+        materialized_grad (bool, optional): If ``True``, set the gradient for unused inputs
+            to zero instead of None. This is useful when computing higher-order derivatives.
+            If ``materialized_grad`` is ``True`` and ``allow_unused`` is ``False``, an error
+            will be raised. Defaults to ``False``.
+
     """
+    if materialized_grad:
+        if allow_unused is False:
+            raise RuntimeError("materialized_grad=True conflicts with allow_unused=False")
```

Comment:
Could we update the error to something like: `Expected allow_unused to be True or not passed when materialize_grad=True, but got: allow_unused=False.`

Reply:
RuntimeError -> ValueError, since we're dealing with a conflict of the arguments